feat: add metaso.cn API search

hu-mengliang opened the pull request:
## Description

Add metaso.cn API search

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have linked this PR to an issue using the Development section on the right sidebar or by adding `Fixes #issue-number` in the PR description (**required**)
- [x] I have checked if any dependencies need to be added or updated in `pyproject.toml` and `uv lock`
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation if needed:
- [x] I have added examples if this is a new feature

If you are unsure about any of these, don't hesitate to ask. We are here to help!

Wendong-Fan reviewed (commented):
thanks @hu-mengliang  for creating this PR! Overall looks good to me, could you refer to https://github.com/camel-ai/camel/blob/master/examples/toolkits/search_toolkit.py and https://github.com/camel-ai/camel/blob/master/test/toolkits/test_search_functions.py to add the example code and test code? Thanks in advance!

hu-mengliang commented:
@Wendong-Fan Thanks for your suggestion. I have updated the `search_metaso` method to add type restrictions for the search range args"scope"and configured the METASO API key in the `.env` file.Also, I have added the example code and test code.
